org.eclipse.lsp4e/src/org/eclipse/lsp4e/progress/LSPProgressManager.java

```diff
@@ -63,7 +67,10 @@ public void connect(final LanguageServer languageServer) {
 			LanguageServerPlugin.logInfo(
 					"Old progress with identifier " + jobIdentifier + " discarded due to new create progress request"); //$NON-NLS-1$//$NON-NLS-2$
 		}
+		return CompletableFuture.runAsync(() -> createJob(languageServer, queue, jobIdentifier));

```

Comment:
I don't think calling `createJob` in a `runAsync()` is a good thing. Jobs are already async, so here we may wait for more time before actually running: the time it takes to the future to be run + the time it takes for the job to be scheduled.
As we do not need the server to wait before returning here (maybe we'll need the server to wait some day on the Job *completion* but not on the Job *creation*), it seems more efficient to call `createJob` and then return directly ` CompletableFuture.completedFuture(null)`